File: frontend/src/types/relatorio.ts
/** TypeScript interfaces for report types matching Pydantic schemas. */

export interface RelatorioFiltro {
  data_inicio: string;
  data_fim: string;
}

export interface MissoesPorStatusItem {
  status: string;
  total: number;
}

export interface MissoesPorStatusResponse {
  data_inicio: string;
  data_fim: string;
  items: MissoesPorStatusItem[];
  total_geral: number;
}

export interface AreaPorClienteItem {
  cliente_id: number;
  cliente_nome: string;
  area_total_pulverizada: number;
}

export interface AreaPorClienteResponse {
  data_inicio: string;
  data_fim: string;
  items: AreaPorClienteItem[];
}

export interface FinanceiroResumoResponse {
  data_inicio: string;
  data_fim: string;
  total_custo_estimado: number;
  total_custo_realizado: number;
  total_valor_faturado: number;
  total_missoes: number;
}

export interface UtilizacaoDroneItem {
  drone_id: number;
  drone_identificacao: string;
  drone_modelo: string;
  horas_voadas_periodo: number;
  total_missoes: number;
}

export interface UtilizacaoDroneResponse {
  data_inicio: string;
  data_fim: string;
  items: UtilizacaoDroneItem[];
}

File: app/schemas/relatorio.py
"""Schemas for operational and financial reports."""

from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class RelatorioFiltro(BaseModel):
    """Common filter for report endpoints."""

    data_inicio: date
    data_fim: date


class MissoesPorStatusItem(BaseModel):
    """Single row in the missions-by-status report."""

    status: str
    total: int


class MissoesPorStatusResponse(BaseModel):
    """Report: missions grouped by status within a period."""

    data_inicio: date
    data_fim: date
    items: list[MissoesPorStatusItem]
    total_geral: int


class AreaPorClienteItem(BaseModel):
    """Single row in the sprayed-area-by-client report."""

    cliente_id: int
    cliente_nome: str
    area_total_pulverizada: Decimal


class AreaPorClienteResponse(BaseModel):
    """Report: total sprayed area by client within a period."""

    data_inicio: date
    data_fim: date
    items: list[AreaPorClienteItem]


class FinanceiroResumoResponse(BaseModel):
    """Report: financial summary within a period (only ENCERRADA_FINANCEIRAMENTE)."""

    data_inicio: date
    data_fim: date
    total_custo_estimado: Decimal
    total_custo_realizado: Decimal
    total_valor_faturado: Decimal
    total_missoes: int


class UtilizacaoDroneItem(BaseModel):
    """Single row in the drone utilization report."""

    drone_id: int
    drone_identificacao: str
    drone_modelo: str
    horas_voadas_periodo: Decimal
    total_missoes: int


class UtilizacaoDroneResponse(BaseModel):
    """Report: drone utilization (flight hours) within a period."""

    data_inicio: date
    data_fim: date
    items: list[UtilizacaoDroneItem]
